packages/engine/Source/Scene/HeightReference.js

```diff
@@ -12,17 +12,74 @@ const HeightReference = {
   NONE: 0,
 
   /**
-   * The position is clamped to the terrain.
+   * The position is clamped to the terrain and 3D Tiles.
    * @type {number}
    * @constant
    */
   CLAMP_TO_GROUND: 1,
 
   /**
-   * The position height is the height above the terrain.
+   * The position height is the height above the terrain and 3D Tiles.
    * @type {number}
    * @constant
    */
   RELATIVE_TO_GROUND: 2,
+
+  /**
+   * The position is clamped to terain.
+   * @type {number}
+   * @constant
+   */
+  CLAMP_TO_TERRAIN: 3,
+
+  /**
+   * The position height is the height above terrain.
+   * @type {number}
+   * @constant
+   */
+  RELATIVE_TO_TERRAIN: 4,
+
+  /**
+   * The position is clamped to 3D Tiles.
+   * @type {number}
+   * @constant
+   */
+  CLAMP_TO_3D_TILE: 5,
+
+  /**
+   * The position height is the height above 3D Tiles.
+   * @type {number}
+   * @constant
+   */
+  RELATIVE_TO_3D_TILE: 6,
 };
+
 export default Object.freeze(HeightReference);
+
+/**
+ * Returns true if the height should be offset relative to the surface
+ * @param {HeightReference} heightReference
+ * @returns true if the height should be offset relative to the surface
+ * @private
+ */
+export function isHeightReferenceClamp(heightReference) {
```

Comment:
This reads a little weird to me and doesn't portray what it means at first glace. Would `isHeightReferenceClamping` be a little more clear? 

Up to you, won't bikeshed

Reply:
I was trying to keep it the same terminology as the enum value. Since this is an internal function and is documented, OK to leave it as is?